chatsky/context_storages/database.py

```diff
@@ -8,181 +8,135 @@
-
     @abstractmethod
-    async def contains_async(self, key: Hashable) -> bool:
+    async def load_field_latest(self, ctx_id: str, field_name: str) -> List[Tuple[Hashable, bytes]]:
         """
-        Asynchronous method for finding whether any Context is stored with given key.
-
-        :param key: Hashable key used to check if Context instance is stored.
-        :return: True if there is Context accessible by given key, False otherwise.
+        Load the latest field data.
         """
         raise NotImplementedError
 
-    def __len__(self) -> int:
-        """
-        Synchronous method for retrieving number of stored Contexts.
-
-        :return: The number of stored Contexts.
-        """
-        return asyncio.run(self.len_async())
-
     @abstractmethod
-    async def len_async(self) -> int:
+    async def load_field_keys(self, ctx_id: str, field_name: str) -> List[Hashable]:
         """
-        Asynchronous method for retrieving number of stored Contexts.
-
-        :return: The number of stored Contexts.
+        Load all field keys.
         """
         raise NotImplementedError
 
-    def get(self, key: Hashable, default: Optional[Context] = None) -> Context:
-        """
-        Synchronous method for accessing stored Context, returning default if no Context is stored with the given key.
-
-        :param key: Hashable key used to store Context instance.
-        :param default: Optional default value to be returned if no Context is found.
-        :return: The stored context, associated with the given key or default value.
-        """
-        return asyncio.run(self.get_async(key, default))
-
-    async def get_async(self, key: Hashable, default: Optional[Context] = None) -> Context:
+    @abstractmethod
+    async def load_field_items(self, ctx_id: str, field_name: str, keys: Set[Hashable]) -> List[bytes]:
         """
-        Asynchronous method for accessing stored Context, returning default if no Context is stored with the given key.
-
-        :param key: Hashable key used to store Context instance.
-        :param default: Optional default value to be returned if no Context is found.
-        :return: The stored context, associated with the given key or default value.
+        Load field items.
         """
-        try:
-            return await self.get_item_async(str(key))
-        except KeyError:
-            return default
+        raise NotImplementedError
 
-    def clear(self):
+    @abstractmethod
+    async def update_field_items(self, ctx_id: str, field_name: str, items: List[Tuple[Hashable, bytes]]) -> None:
         """
-        Synchronous method for clearing context storage, removing all the stored Contexts.
+        Update field items.
         """
-        return asyncio.run(self.clear_async())
+        raise NotImplementedError
 
     @abstractmethod
-    async def clear_async(self):
+    async def delete_field_keys(self, ctx_id: str, field_name: str, keys: List[Hashable]) -> None:
```

Comment:
Add method for deleting a context entirely (`delete_main_info` + `delete_field_keys`).

Reply:
Maybe rename it to `delete_context` then? Not very clear that deleting main info also deletes all info.